File: repos_estudiantes_TARQUI_primera_entrega/Quispe_Mollisaca_Jimmy_Brandon/Polimorfismo/Ejer2/Restaurante.java
package Polimorfismo;
class Cocinero {
    String nombre;
    int sueldoMes;
    int horasExtra;
    float sueldoHora;

    public Cocinero(String nombre, int sueldoMes, int horasExtra, float sueldoHora) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.horasExtra = horasExtra;
        this.sueldoHora = sueldoHora;
    }

    public float sueldoTotal() {
        return sueldoMes + (horasExtra * sueldoHora);
    }

    public void mostrarInfo() {
        System.out.println("Cocinero [nombre=" + nombre + ", sueldoMes=" + sueldoMes +
                ", horasExtra=" + horasExtra + ", sueldoHora=" + sueldoHora + "]");
    }
}

class Mesero {
    String nombre;
    int sueldoMes;
    int horasExtra;
    float sueldoHora;
    float propina;

    public Mesero(String nombre, int sueldoMes, int horasExtra, float sueldoHora, float propina) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.horasExtra = horasExtra;
        this.sueldoHora = sueldoHora;
        this.propina = propina;
    }

    public float sueldoTotal() {
        return sueldoMes + (horasExtra * sueldoHora) + propina;
    }

    public void mostrarInfo() {
        System.out.println("Mesero [nombre=" + nombre + ", sueldoMes=" + sueldoMes +
                ", horasExtra=" + horasExtra + ", sueldoHora=" + sueldoHora + ", propina=" + propina + "]");
    }
}

class Administrativo {
    String nombre;
    float sueldoMes;
    String cargo;

    public Administrativo(String nombre, float sueldoMes, String cargo) {
        this.nombre = nombre;
        this.sueldoMes = sueldoMes;
        this.cargo = cargo;
    }

    public float sueldoTotal() {
        return sueldoMes;
    }

    public void mostrarInfo() {
        System.out.println("Administrativo [nombre=" + nombre + ", sueldoMes=" + sueldoMes +
                ", cargo=" + cargo + "]");
    }
}

public class Restaurante {
    public static void mostrarPorSueldo(Object[] empleados) {
        System.out.println("Información de todos los empleados:");
        for (Object empleado : empleados) {
            if (empleado instanceof Cocinero) {
                ((Cocinero) empleado).mostrarInfo();
            } else if (empleado instanceof Mesero) {
                ((Mesero) empleado).mostrarInfo();
            } else if (empleado instanceof Administrativo) {
                ((Administrativo) empleado).mostrarInfo();
            }
        }
    }

    public static void mostrarPorSueldo(float sueldo, Object[] empleados) {
        System.out.println("Empleados con sueldo mensual igual a " + sueldo + ":");
        for (Object empleado : empleados) {
            if (empleado instanceof Cocinero && ((Cocinero) empleado).sueldoMes == sueldo) {
                ((Cocinero) empleado).mostrarInfo();
            } else if (empleado instanceof Mesero && ((Mesero) empleado).sueldoMes == sueldo) {
                ((Mesero) empleado).mostrarInfo();
            } else if (empleado instanceof Administrativo && ((Administrativo) empleado).sueldoMes == sueldo) {
                ((Administrativo) empleado).mostrarInfo();
            }
        }
    }

    public static void main(String[] args) {
    	
        Cocinero cocinero = new Cocinero("Carlos", 2000, 10, 15.5f);
        Mesero mesero1 = new Mesero("Luis", 1500, 5, 10.0f, 300.5f);
        Mesero mesero2 = new Mesero("Juan", 1800, 3, 8.0f, 200.0f);
        Administrativo admin1 = new Administrativo("Ana", 2500.0f, "Gerente");
        Administrativo admin2 = new Administrativo("Pedro", 1500.0f, "Supervisor");

        Object[] empleados = {cocinero, mesero1, mesero2, admin1, admin2};

        mostrarPorSueldo(empleados);

        mostrarPorSueldo(1500, empleados);
    }
}

File: repos_estudiantes_TARQUI_primera_entrega/Quispe_Mollisaca_Jimmy_Brandon/Polimorfismo/Ejer2/Restaurante.py
class Cocinero:
    def __init__(self, nombre, sueldoMes, horasExtra, sueldoHora):
        self.nombre = nombre
        self.sueldoMes = sueldoMes
        self.horasExtra = horasExtra
        self.sueldoHora = sueldoHora

    def sueldo_total(self):
        return self.sueldoMes + (self.horasExtra * self.sueldoHora)

    def mostrar_info(self):
        print(f"Cocinero [nombre={self.nombre}, sueldoMes={self.sueldoMes}, horasExtra={self.horasExtra}, sueldoHora={self.sueldoHora}]")


class Mesero:
    def __init__(self, nombre, sueldoMes, horasExtra, sueldoHora, propina):
        self.nombre = nombre
        self.sueldoMes = sueldoMes
        self.horasExtra = horasExtra
        self.sueldoHora = sueldoHora
        self.propina = propina

    def sueldo_total(self):
        return self.sueldoMes + (self.horasExtra * self.sueldoHora) + self.propina

    def mostrar_info(self):
        print(f"Mesero [nombre={self.nombre}, sueldoMes={self.sueldoMes}, horasExtra={self.horasExtra}, sueldoHora={self.sueldoHora}, propina={self.propina}]")


class Administrativo:
    def __init__(self, nombre, sueldoMes, cargo):
        self.nombre = nombre
        self.sueldoMes = sueldoMes
        self.cargo = cargo

    def sueldo_total(self):
        return self.sueldoMes

    def mostrar_info(self):
        print(f"Administrativo [nombre={self.nombre}, sueldoMes={self.sueldoMes}, cargo={self.cargo}]")


class Restaurante:
    @staticmethod
    def mostrar_por_sueldo(empleados):
        print("Información de todos los empleados:")
        for empleado in empleados:
            empleado.mostrar_info()

    @staticmethod
    def mostrar_por_sueldo_filtrado(sueldo, empleados):
        print(f"Empleados con sueldo mensual igual a {sueldo}:")
        for empleado in empleados:
            if isinstance(empleado, Cocinero) and empleado.sueldoMes == sueldo:
                empleado.mostrar_info()
            elif isinstance(empleado, Mesero) and empleado.sueldoMes == sueldo:
                empleado.mostrar_info()
            elif isinstance(empleado, Administrativo) and empleado.sueldoMes == sueldo:
                empleado.mostrar_info()


# Instanciar empleados y ejecutar métodos
if __name__ == "__main__":
    cocinero = Cocinero("Carlos", 2000, 10, 15.5)
    mesero1 = Mesero("Luis", 1500, 5, 10.0, 300.5)
    mesero2 = Mesero("Juan", 1800, 3, 8.0, 200.0)
    admin1 = Administrativo("Ana", 2500.0, "Gerente")
    admin2 = Administrativo("Pedro", 1500.0, "Supervisor")

    empleados = [cocinero, mesero1, mesero2, admin1, admin2]

    Restaurante.mostrar_por_sueldo(empleados)
    Restaurante.mostrar_por_sueldo_filtrado(1500, empleados)